src/openrct2/ride/Vehicle.cpp

```diff
@@ -7596,7 +7601,9 @@ bool Vehicle::UpdateTrackMotionForwards(CarEntry* carEntry, Ride* curRide, rct_r
             auto brakeSpeed = brake_speed << 16;
             if (brakeSpeed < _vehicleVelocityF64E08)
             {
-                acceleration = -_vehicleVelocityF64E08 * 16;
+                auto trackElement = MapGetTrackElementAtOfTypeSeq(TrackLocation, trackType, 0);
+                if ((trackElement != nullptr && trackElement->AsTrack()->GetBrakeClosed()) || trackElement == nullptr)
```

Comment:
this is odd? Why care about the nullptr case?

Reply:
If the vehicle believes it is on a brake piece but the track element cannot be found to verify the brake is closed, assume the brake is closed